core/src/main/java/org/apache/accumulo/core/fate/Fate.java

```diff
@@ -458,10 +458,14 @@ public Exception getException(FateId fateId) {
     }
   }
 
+  private boolean isExecutorAlive(){
+    return executor != null && !executor.isTerminated();
+  }
+
   /**
    * Initiates shutdown of background threads and optionally waits on them.
    */
-  public void shutdown(boolean waitForBackgroundThreads) {
+  public void shutdown(long timeout, TimeUnit timeUnit) {
     if (keepRunning.compareAndSet(true, false)) {
       fatePoolWatcher.shutdown();
       if (executor != null) {
```

Comment:
This is unrelated to your PR but something I noticed is executor should never be null as it's final and created in the constructor.

Reply:
did not realize that.  Removed the null checks and it cleaned up the code.

Reply:
[fd5b2a5](https://github.com/apache/accumulo/pull/4248/commits/fd5b2a5d22ab71a9195b5f1aba80ae8201be836e)